### PR TITLE
[#32] Added ECMP support for OSPF

### LINKED ISSUE #32: Enhance ESG code to enable ECMP mode
I would like to request a feature to enable ECMP mode on an Edge - not sure if the preferred method would be through the nsx_edge_router module, or through the respective routing protocol modules (i.e. nsx_ospf)

### PR DESCRIPTION
**Pull request by COvirtNetwork**
Changed file nsx_ospf.py to add a new parameter for ecmp
Closes #32

### CODE REVIEW

File: library/nsx_ospf.py [hunk 1]
```diff
@@ -69,6 +69,15 @@ def check_router_id(current_config, router_id):
         current_config['routing']['routingGlobalConfig']['routerId'] = router_id
         return True, current_config
 
+def check_ecmp(current_config, ecmp):
```
Comment (yfauser) [inline library/nsx_ospf.py:72]: Very Minor NIT; Can you check PEP8 compliance? In this case PEP8 is complaining that there is only one blank line before the new function

**Comment by yfauser** [comment]
@jtompkins81 Very good addition! I could merge it, but wouldn't it be better as a part of nsx_edge_router.py and nsx_dlr.py? 
The ECMP setting is more a global routing setting, and ECMP can be used with static routes & BGP too. Could you look at moving the code to those modules?

**Comment by COvirtNetwork** [comment]
@yfauser I originally intended to add it to nsx_edge_router.py and nsx_dlr.py, then saw in the API guide that the ECMP status cannot be set when deploying an edge, and has to be done by a subsequent call after the edge is already deployed.  I tried to use the router id as a reference, since it is also set globally, independent of what routing protocol will be used.  I saw it was included in the OSPF module, so I took the same approach with enabling ECMP.  What are your thoughts on the most appropriate way to handle this?  Is there a way to build the functionality into the nsx_edge_router.py and nsx_dlr.py code that makes a second API call after the edge VM is deployed?

**Comment by yfauser** [comment]
@jtompkins81 You are right, the router id is one of those cases of 'in between' too.
I'm leaving it up to you to decide wether you want to leave it in the ospf module or move it to the create modules.

The create modules are actually first creating the ESG/DLR and then checking and changing their state after they are deployed. I needed to do this in case there are changes to the ESG/DLR after deployment. 

This is the create part of the ESG:
```python
    if module.params['state'] == 'present':
        if not edge_id:
            edge_id, edge_params = create_edge_service_gateway(client_session, module)
            changed = True
    elif module.params['state'] == 'absent':
        if edge_id:
            esg_delete_response = delete_edge_service_gateway(client_session, edge_id)
            module.exit_json(changed=True, esg_create_response=esg_create_response,
                             esg_delete_response=esg_delete_response)
        else:
            module.exit_json(changed=False, esg_create_response=esg_create_response,
                             esg_delete_response=esg_delete_response)
```

And those are the state checks:

```python
routes, current_dfgw = get_esg_routes(client_session, edge_id)
fw_state = get_firewall_state(client_session, edge_id)
ifaces_changed = check_interfaces(client_session, edge_id, module)
routes_changed = check_routes(client_session, edge_id, routes, module)
```

It should be relatively straightforward to port your code to the second part. It's just that in those modules I still used my 'old way' of transporting the changed state to the end of the module. The 'newer' ospf module has an easier way to do so. The code style evolved along the way. 

Again, I'm also fine leaving it in the ospf module - just let me know what you prefer.

**Pull request merged**